File: 2527-count-subarrays-with-fixed-bounds/count-subarrays-with-fixed-bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        long ans=0;
        int start=0,minstart=0,maxstart=0;
        boolean minf=false,maxf=false;
        int n=nums.length;

        for(int i=0;i<n;i++){
            if(nums[i]<minK || nums[i]>maxK){
                minf=false;
                maxf=false;
                start=i+1;
            }
            else{
                if(nums[i]==minK){
                    minf=true;
                    minstart=i;
                }
                if(nums[i]==maxK){
                    maxf=true;
                    maxstart=i;
                }
            }
            if(minf && maxf){
                ans+=Math.min(minstart,maxstart)-start+1;
            }
        }
        return ans;
    }
}

File: 2527-count-subarrays-with-fixed-bounds/count-subarrays-with-fixed-bounds.cpp
class Solution {
    using ll=long long;
public:
    long long countSubarrays(vector<int>& nums, int minK, int maxK) {
        ll ans=0;
        ll s=0,mni=-1,mxi=-1;

        for(ll i=0;i<nums.size();i++){
            if(nums[i]<minK || nums[i]>maxK){
                s=i+1;
                mni=-1,mxi=-1;
            }
            else{
                if(nums[i]==minK){
                    mni=i;
                }
                if(nums[i]==maxK){
                    mxi=i;
                }
            }
            if(mni!=-1 && mxi!=-1){
                ans+=min(mni,mxi)-s+1;
            }
        }
        return ans;
    }
};